File: rules/python/tests/bind-all-interfaces.py
# Fixture: bind-all-interfaces
from flask import Flask
import uvicorn

app = Flask(__name__)

# match: Flask exposed on all interfaces
app.run(host="0.0.0.0", port=8080)

# match: uvicorn keyword form
uvicorn.run("main:app", host="0.0.0.0")

# no-match: loopback only
app.run(host="127.0.0.1", port=8080)

# no-match: no host argument
app.run(port=5000)

File: rules/javascript/tests/bind-all-interfaces.js
// Fixture: bind-all-interfaces (js)
const http = require("http");
const server = http.createServer();

// match: listen to 0.0.0.0
server.listen(3000, "0.0.0.0");

// match: { host: '0.0.0.0' } option
const opts = { host: "0.0.0.0" };

// no-match: loopback
server.listen(3000, "127.0.0.1");

// no-match: unrelated host
const meta = { host: "api.example.com" };
